### ISSUE TITLE
[BUG] catkin_make tool fails while building objects

### ISSUE THREAD

**Issue by marcoluis97** [project-aslan/Aslan#44]
While following the installation guide for the Project Aslan Software, I've come across a build error after executing the command:

`catkin_make`

The errors can be observed in the following screenshot

![build_catkin_error](https://user-images.githubusercontent.com/94386623/141816317-47dc2a18-2ee2-4ae1-861a-401fb3f4e211.png)

As I've said, I've executed the installation according to the ProjectAslan Instalation guideline on the README.md page. How should I proceed in order to correctly build the code?

**System**
 - OS: Ubuntu 18.04
 - ASLAN Version: 1.2.0
 - ROS Version: melodic
 - Source Code

[comment]: # (In the case of urgent feedback needed, assign this issue to yourself and one of the core developers:)
[comment]: # (EfimiaPanagiotaki-StreetDrone)
[comment]: # (AbdelrahmanBarghouth-StreetDrone)
[comment]: # (FionanOSullivan-StreetDrone)

**Comment by SanderVanDijk-StreetDrone** [comment]
Hi @marcoluis97 thanks for the report. The error you see is the compiler actually crashing, rather than a regular compilation error. This usually means that your machine is running out of resources, probably memory specifically. How much memory do you have on the machine that you are building on?

You see at the end that `catkin_make` runs `make -j8 -l8` underneath, which means it runs 8 compilation jobs in parallel. You can reduce this amount to try to not have too many use too much memory at the same time, by running:

```
catkin_make --make-args -j4
```

Keep reducing the number, down to 1, if you still get the same issue.

**Comment by marcoluis97** [comment]
Hi @SanderVanDijk-StreetDrone thanks for the answer. Yes, after some research I ended up at the same conclusion and tried to run a similar command. I was able to run Aslan and put the simulation environment running but am now encountering a few configuration problems. Could it be from the first failed build of the catkin workspace for Aslan? Thank you once again. 
I've noticed as well that the installation guide you provide does not include the need to source the catkin build tool and add it to the .bashrc file. Should that be necessary in order to work with multiples workspaces down the line?

**Comment by SanderVanDijk-StreetDrone** [comment]
It should build everything properly after retrying, but to be certain you can try deleting the `build` and `devel` directories and build from scratch again, though that will take some time. If you then still have any problems feel free to open a new issue for them!

When you create a new workspace to work on top of Aslan you indeed need to follow the regular ROS steps to do so, including sourcing the right workspaces. Doing that in `.bashrc` would indeed be the easiest way.
